src/coreclr/nativeaot/Runtime/threadstore.inl

```diff
@@ -25,16 +25,6 @@ inline Thread * ThreadStore::GetCurrentThread()
     return pCurThread;
 }
 
-// static
-inline Thread * ThreadStore::GetCurrentThreadIfAvailable()
```

Comment:
We do not compile with link time codegen enabled currently (we would have to ship all runtime C/C++sources in the package in order to do that). It means that explicitly inlining methods like this is more important here than in coreclr (coreclr.dll compiles with link time codegen).

Is this method called on any perf critical paths that will regress once it stops being inlined?

Reply:
I think most perf-sensitive cases will call `GetCurrentThread` if they know the thread is available or `GetCurrentThreadRaw` if unsure.

The reason this cannot be `inline` is that we fail on Linux at linking stage with missing method when we call this from `PalHijack`. In the context of `PalHijack` inline would not be too important as we are making syscalls which are heavier things.

Perhaps I can just make a sibling method that is not inlined - just to call from `PalHijack.

Reply:
> The reason this cannot be inline is that we fail on Linux at linking stage with missing method when we call this from PalHijack.

You should be able to fix that by `#including` `thread.inl` in the PAL implementation file.

Reply:
Somehow did not think about that.